### DISCUSSION TITLE
Сопряжение с Tion 3s

### DISCUSSION THREAD

**Discussion by MagFam** [TionAPI/HA-tion#121] [General]
Добрый день!
Не получается подключиться к Tion 3S, при сопряжении появляется ошибка "Could not pair!"
Ранее подключенные к бризеру пульты и телефоны отключены.
Подскажите, в какую сторону копать? )

**Система:** 

 "installation_type": "Home Assistant OS",
    "version": "2023.2.5",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.7",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Moscow",
    "os_name": "Linux",
    "os_version": "5.15.90",
    "supervisor": "2023.01.1",
    "host_os": "Home Assistant OS 9.5",
    "docker_version": "20.10.22",
    "chassis": "embedded",
    "run_as_root": true

**При этом, если посмотреть диагностические данные Bluetooth-адаптера, то видно, что бризер определяется:**

"connectable_history": [
        {
          "name": "Tion Breezer 3S",
          "address": "E9:7F:95:84:11:A8",
          "rssi": -81,
          "manufacturer_data": {},
          "service_data": {},
          "service_uuids": [],
          "source": "80:00:0B:D3:38:BB",
          "advertisement": [
            "Tion Breezer 3S",
            {},
            {},
            [],
            null,
            -81,
            [
              "/org/bluez/hci0/dev_E9_7F_95_84_11_A8",
              {
                "Address": "E9:7F:95:84:11:A8",
                "AddressType": "random",
                "Name": "Tion Breezer 3S",
                "Alias": "Tion Breezer 3S",
                "Paired": false,
                "Trusted": false,
                "Blocked": false,
                "LegacyPairing": false,
                "RSSI": -81,
                "Connected": false,
                "UUIDs": [],
                "Adapter": "/org/bluez/hci0",
                "ServicesResolved": false,
                "AdvertisingFlags": {
                  "__type": "<class 'bytearray'>",
                  "repr": "bytearray(b'\\x04')"
                }
              }
            ]
          ],
          "device": {
            "__type": "<class 'bleak.backends.device.BLEDevice'>",
            "repr": "BLEDevice(E9:7F:95:84:11:A8, Tion Breezer 3S)"
          },
          "connectable": true,
          "time": 1289478.908745685
        }
      ], 

**также есть лог интеграции Bluetooth Low Energy Monitor:**

2023-03-07 23:03:03.647 DEBUG (MainThread) [custom_components.ha_tion_btle.config_flow] Showing pair info
2023-03-07 23:03:17.024 DEBUG (MainThread) [custom_components.ha_tion_btle.config_flow] Real pairing step
2023-03-07 23:03:17.025 DEBUG (MainThread) [custom_components.ha_tion_btle.config_flow] {'model': 'S3', 'name': 'Tion Breezer', 'mac': 'E9:7F:95:84:11:A8', 'keep_alive': 60, 'away_temp': 15, 'pair': True}
2023-03-07 23:03:17.025 CRITICAL (MainThread) [custom_components.ha_tion_btle.config_flow] getTion: Could not find device with mac='E9:7F:95:84:11:A8'
2023-03-07 23:03:17.032 ERROR (MainThread) [custom_components.ha_tion_btle.config_flow] Cannot pair and get data. Data is {'model': 'S3', 'name': 'Tion Breezer', 'mac': 'E9:7F:95:84:11:A8', 'keep_alive': 60, 'away_temp': 15, 'pair': True}, result is {}; BleakError: Could not find device with mac='E9:7F:95:84:11:A8'
2023-03-07 23:03:20.868 DEBUG (MainThread) [custom_components.ha_tion_btle.config_flow] Pair failed. Returning to first step

**Comment by IATkachenko** [comment]
Доброго дня.

BT нормально в HA проброшен? Я с Home Assistant OS никогда не работал... НА там живет прямо в системе или в контейнере?
Режим сопряжения на бризере включается нормально?

Попробуйте в системе еще сделать обычный BT pairing (через bluetoothctl, например), хотя если бризер в режиме сопряжения, то он должен отвечать вообще на все...

Ну и Вики может быть полезна https://github.com/TionAPI/HA-tion/wiki/Bluetooth

**Reply by MagFam** [reply]
> Доброго дня.
> 
> BT нормально в HA проброшен? Я с Home Assistant OS никогда не работал... НА там живет прямо в системе или в контейнере? Режим сопряжения на бризере включается нормально?
> 
> Попробуйте в системе еще сделать обычный BT pairing (через bluetoothctl, например), хотя если бризер в режиме сопряжения, то он должен отвечать вообще на все...
> 
> Ну и Вики может быть полезна https://github.com/TionAPI/HA-tion/wiki/Bluetooth

Добрый день!
Я особо не разбираюсь в работе HA OS, полагаю HA работает в контейнере, судя по логам:
   "os_name": "Linux",
    "os_version": "5.15.90",
    "supervisor": "2023.03.1",
    "host_os": "Home Assistant OS 9.5",
    **"docker_version": "20.10.22",**
    "chassis": "embedded",
    "run_as_root": true

Бризер сопрягается с другими устройствами, проверял на телефоне. После проверки отсоединил телефон и отключил пульт бризера, как указано в инструкции.

Попробовал Bluetooth стик с отдельной антенной - не помогло.

Через bluetoothctl тоже не получилось подключить бризер.
![image](https://user-images.githubusercontent.com/124466011/225421716-c1caab7f-c2fc-4e21-ad07-e143bf4db196.png)

В вики рекомендации, если я правильно понял, в основном касаются не HA OS.

**Comment by IATkachenko** [comment]
Фигурные скобки вроде лишние.

Да, мы в основном в контейнерах живем. Но все что касается `bluetoothctl` -- применимо и для вас. 
Собственно вот [эта часть](https://github.com/TionAPI/HA-tion/wiki/Bluetooth#%D1%81%D0%BE%D0%BF%D1%80%D1%8F%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5-%D1%83%D1%81%D0%BF%D0%B5%D1%88%D0%BD%D0%BE-%D0%BD%D0%BE-%D0%BD%D0%B5-%D0%BF%D0%BE%D0%BB%D1%83%D1%87%D0%B0%D0%B5%D1%82%D1%81%D1%8F-%D0%BF%D0%BE%D0%BB%D1%83%D1%87%D0%B8%D1%82%D1%8C-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D0%B5) наиболее интересна/полезна

**Reply by MagFam** [reply]
> Фигурные скобки вроде лишние.
> 
> Да, мы в основном в контейнерах живем. Но все что касается `bluetoothctl` -- применимо и для вас. Собственно вот [эта часть](https://github.com/TionAPI/HA-tion/wiki/Bluetooth#%D1%81%D0%BE%D0%BF%D1%80%D1%8F%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5-%D1%83%D1%81%D0%BF%D0%B5%D1%88%D0%BD%D0%BE-%D0%BD%D0%BE-%D0%BD%D0%B5-%D0%BF%D0%BE%D0%BB%D1%83%D1%87%D0%B0%D0%B5%D1%82%D1%81%D1%8F-%D0%BF%D0%BE%D0%BB%D1%83%D1%87%D0%B8%D1%82%D1%8C-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D0%B5) наиболее интересна/полезна

Добрый день!
Получил доступ к хосту. Хочу посмотреть файл /var/lib/bluetooth/МАС-адрес-адаптера/MAC-адрес-бризера, но у меня нет каталога  /var/lib/bluetooth


![Screenshot 2023-03-17 234923](https://user-images.githubusercontent.com/124466011/226041373-aeef9836-8617-44b2-b8b8-616b981b08e8.jpg)

**Reply by IATkachenko** [reply]
Это точно хост, а не контейнер, как описано в ссылке инструкции?

Мой /var/lib, например, выглядит так (но я никогда не видел как оно в HA OS):
```
igor@home:~ $ ls /var/lib/
alsa  apt  bluetooth  containerd  dbus  dhcp  dhcpcd  docker  dpkg  git  logrotate  man-db  misc  
  NetworkManager  nfs  pam  polkit-1  private  python  raspberrypi  samba  sudo  systemd  ucf  
  udisks2  usb_modeswitch  vim
```

В чем суть того что нужно сделать и того что происходит:
За общее взаимодействие по BLE отвечает хост. Нам нужно сделать две вещи:
1. сопряжение по bluetooth. Вы показывали консоль с `bluetoothctl`. Попробуйте в начале простой вариант: 
    * перевести бризер в режим сопряжения
    * запустить bluetoothctl от root'a (`sudo bluetoothctl`)
      * начнется спам из-за интеграции Bluetooth (кстати, она у вас добавлена?)
    * перевести бризер в режим сопряжения
    * выполнить в bluetoothctl `pair E9:7F:95:84:11:A8`
    * потом можно еще и `info E9:7F:95:84:11:A8`
2. сопряжение Tion'a. Это можно сделать через интеграцию. Если BT сопряжение прошло нормально проблем не будет.
 
В вашем исходном логе видно что интеграция чрез bluetooth средства HA не может добраться до бризера. 
Полный путь тут такой: интеграция -> BT в HA (интеграция [bluetooth ](https://www.home-assistant.io/integrations/bluetooth/)+ `bleak`) -> BT в  контейнере (через docker) -> BT на хосте (bluetoothd) -> адаптер -> бризер.

Проверка ключей сопряжения -- проверка последней мили это цепочки.
